front/src/common/RollingStockSelector/RollingStockModal.tsx

```diff
@@ -88,55 +88,62 @@ function RollingStockModal({ ref2scroll }: RollingStockModal) {
     elec: false,
     thermal: false,
   });
-  const [isLoading, setIsLoading] = useState(false);
+  const [rollingStocks, setRollingStocks] = useState<LightRollingStock[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
   const [openRollingStockCardId, setOpenRollingStockCardId] = useState(rollingStockID);
   const { closeModal } = useContext(ModalContext);
 
   if (darkmode) {
     import('./RollingStockDarkMode.scss');
   }
 
-  const { rollingStocks, isSuccess, isError, error } = osrdEditoastApi.useGetLightRollingStockQuery(
-    {
-      pageSize: 1000,
-    },
-    {
-      selectFromResult: (response) => ({
-        ...response,
-        rollingStocks: sortBy(response.data?.results, ['metadata.reference', 'name']) || [],
-      }),
-    }
-  );
-  const [filteredRollingStockList, setFilteredRollingStockList] = useState<LightRollingStock[]>(
-    () => filterRollingStocks(rollingStocks, filters)
+  const { apiRollingStocks, isError, error, requestIsLoading } =
```

Comment:
Could you have instantiated the loading like `isLoading: requestIsLoading` instead of declaring it in selectFromResult ?

Reply:
RTK already gives a isLoading variable, but I can't use it because at the mounting of the component, it is false (but I need it to be true). That is why I had to create a custom requestIsLoading :cry:
I don't find it very clean, if you have another solution, I would be glad to improve the code ! :relaxed:

Reply:
It seems that there is also an isUninitialized params, did you try it ? Maybe when isUninitialized goes to false, is Loading is true ? But would it be more elegant than your solution ? Not sure :)

Reply:
Oh thanks, yes I think it would be more elegant ! I need to check also what @anisometropie tried ! :D